### PR TITLE
Implement couples of APIs directly without  GenericOperation

### PR DESCRIPTION
**Pull request by Crazykev**
implement couples of APIs directly without  GenericOperation.

/cc @laijs PTAL

Signed-off-by: Crazykev <crazykev@zju.edu.cn>

### CODE REVIEW

File: hypervisor/vm.go [hunk 1]
```diff
@@ -355,10 +357,12 @@ func (vm *Vm) AddNic(info *api.InterfaceDescription) error {
 }
 
 func (vm *Vm) DeleteNic(id string) error {
+	if vm.ctx == nil || vm.ctx.current != StateRunning {
```
Comment (laijs) [inline hypervisor/vm.go]: after #432, we don't need to test 'vm.ctx == nil' anymore, please remove it

File: hypervisor/vm.go [hunk 2]
```diff
@@ -628,30 +640,23 @@ func (vm *Vm) Pause(pause bool) error {
...
-		if ctx.PauseState == PauseStateBusy {
-			result <- fmt.Errorf("%s fails: earlier Pause or Unpause operation has not finished", command)
-			return
-		} else if ctx.PauseState == pauseState {
-			result <- nil
-			return
-		}
+	var err error
+	oldPauseState = ctx.PauseState
+	if ctx.PauseState == PauseStateBusy {
```
Comment (laijs) [inline hypervisor/vm.go]: the old code is in statemachine, so we have PauseStateBusy for state-checking and synchronization. could you add ctx.pauseLock to protect the Pause(), and remove PauseStateBusy. (also move the changes of Pause() out from this PR, create a new one with my suggestion)

File: hypervisor/vm.go (hunk 1 above)
Comment (Crazykev) [inline hypervisor/vm.go]: @laijs fixed.

File: hypervisor/vm.go (hunk 2 above)
Comment (Crazykev) [inline hypervisor/vm.go]: @laijs I've reverted changes about Pause, will file another PR later.

**Comment by laijs** [comment]
LGTM

**Pull request merged**